### PR TITLE
 Add developer apps page, call new SDK developer apps + grants APIs [C-2624]

### PR DESCRIPTION
**Pull request by nicoback2**
### Description
Just a draft bc need to bump SDK ver (should we make it major bc of walletAPI -> auth change? (@rickyrombo @sliptype)

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?
Tested via browser and linking local SDK on this branch:https://github.com/AudiusProject/audius-protocol/pull/5150 + changing the entity manager configs in SDK manually

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/web/src/pages/developer-page/DeveloperPage.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      </button>
+      <div>
+        {createAppResult == null ? null : (
+          <div>
+            <div>
+              <p>Success! Here are your app&apos;s credentials:</p>
+              <p>
+                API Key: {createAppResult.apiKey}
+                <br />
+                API Secret: {createAppResult.apiSecret}
```
Comment (rickyrombo) [inline packages/web/src/pages/developer-page/DeveloperPage.tsx:128]: I assume we'll have design at some point around this but thought I'd mention anyways that this probably makes sense to put in a password field

File: packages/web/src/pages/App.js [hunk 2]
```diff
@@ -812,7 +814,7 @@ class App extends Component {
                 isMobile={isMobileClient}
                 render={() => <SettingsPage subPage={SubPage.ABOUT} />}
               />
-
+              <Route exact path={DEVELOPER_PAGE} component={DeveloperPage} />
```
Comment (rickyrombo) [inline packages/web/src/pages/App.js:817]: should we feature flag this?

File: packages/web/src/pages/developer-page/DeveloperPage.tsx (hunk 1 above)
Comment (nicoback2) [reply packages/web/src/pages/developer-page/DeveloperPage.tsx:128]: oh yes 100%

File: packages/web/src/pages/App.js (hunk 2 above)
Comment (nicoback2) [reply packages/web/src/pages/App.js:817]: yea good call

File: packages/web/src/pages/developer-page/DeveloperPage.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+const createGrant = async ({ userId, appApiKey }: CreateGrantRequest) => {
+  const sdk = await audiusSdk()
+  await sdk.grants.createGrant({
+    userId,
+    appApiKey
+  })
+  return appApiKey
+}
+
+// TODO: Move to saga, add confirmation (unless we confirm in SDK itself)
```
[the remark this answers is not in the document]
Comment (sliptype) [reply packages/web/src/pages/developer-page/DeveloperPage.tsx:26]: Nah, audius-query is read only for now

File: packages/web/src/pages/developer-page/DeveloperPage.tsx [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        }
+      }
+      setCreateAppResult(res)
+    },
+    [currentUserId]
+  )
+
+  const handleDeleteDeveloperApp = useCallback(async () => {
+    if (!currentUserId || !createAppResult) return
+    try {
```
Comment (sliptype) [inline packages/web/src/pages/developer-page/DeveloperPage.tsx:101]: Should we call `reset()` in here too? And I assume we are going to be fetching the apps via a discovery endpoint right

**Review by sliptype — APPROVED** [review]
Good stuff! Thanks for updating the sdk usages to match the new interface 🙏

File: packages/web/src/pages/developer-page/DeveloperPage.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (rickyrombo) [reply packages/web/src/pages/developer-page/DeveloperPage.tsx:26]: 😢

File: packages/web/src/pages/developer-page/DeveloperPage.tsx (hunk 4 above)
Comment (nicoback2) [reply packages/web/src/pages/developer-page/DeveloperPage.tsx:101]: Nah we can leave the state as is until they create a new key pair (this is just for demoing anyway)

/ yes we will!

**Comment by audius-infra** [comment]
It looks like there may be some changes to native mobile code, which requires triggering a full app release. Please follow the instructions here: https://www.notion.so/audiusproject/When-to-bump-app-version-2644a8f772364a4d91f44abcba44ce0b?pvs=4. cc @nicoback2 @sliptype

**Pull request merged**